src/librustc_codegen_llvm/mir/analyze.rs

```diff
@@ -152,44 +154,46 @@ impl Visitor<'tcx> for LocalAnalyzer<'mir, 'a, 'll, 'tcx> {
 
     fn visit_place(&mut self,
                    place: &mir::Place<'tcx>,
-                   context: PlaceContext<'tcx>,
+                   mut context: PlaceContext<'tcx>,
                    location: Location) {
         debug!("visit_place(place={:?}, context={:?})", place, context);
         let cx = self.fx.cx;
+        let neo_place = cx.tcx.as_new_place(place);
+
+        let mut base_ty = neo_place.base.ty(self.fx.mir);
 
-        if let mir::Place::Projection(ref proj) = *place {
+        for elem in neo_place.elems.iter() {
+            let elem_ty = PlaceTy::from(base_ty).projection_ty(cx.tcx, &elem).to_ty(cx.tcx);
```

Comment:
I'm not sure yet if it matters, but -- this code isn't visiting things in the same order as it used to do. It seems like it **might** matter. For example, if we have `a.b.c` and `c` is a zero-sized field, then the old code would stop recursing when looking at `c` and never visit `a.b`.

This code -- iiuc -- will first process `a.b` and then extend to `c`.